src/go/rpk/pkg/cli/cmd/cluster/maintenance/enable.go

```diff
@@ -66,7 +70,48 @@ node exists that is already in maintenance mode then an error will be returned.
 
 			out.MaybeDie(err, "error enabling maintenance mode for node %d: %v", nodeId, err)
 			fmt.Printf("Successfully enabled maintenance mode for node %d\n", nodeId)
+
+			if !wait {
+				return
+			}
+
+			fmt.Println("Waiting for node to drain...")
+
+			var table *out.TabWriter
+			retries := 3
+			for {
+				b, err := client.Broker(nodeId)
+				if err == nil && b.Maintenance == nil {
+					err = fmt.Errorf("maintenance mode not supported. upgrade in progress?")
```

Comment:
because it is an anomalous situation that shouldn't occur. in fact, it shouldn't occur any time in the future after an upgrade to v22.1.1. but the apis don't specifically guarantee non-nil, so it's mostly about writing code that handles all possible cases.